// mongo-lite/lib/driver.js
// Generated by CoffeeScript 1.4.0
(function() {
  var Driver, helper, _,
    __slice = [].slice;

  module.exports = Driver = {};

  _ = require('underscore');

  helper = require('./helper');

  _(Driver).extend({
    options: {
      safe: false,
      multi: false
    },
    extendedOptions: {
      generateId: false,
      convertId: false,
      perPage: 25,
      maxPerPage: 100
    },
    useHandyButNotStandardDefaults: function() {
      _(this.options).extend({
        safe: true,
        multi: true
      });
      return _(this.extendedOptions).extend({
        generateId: true,
        convertId: true
      });
    },
    connect: function() {
      var args, collectionNames, connection, db, dbOptions, name, url, _i, _len;
      url = arguments[0], args = 2 <= arguments.length ? __slice.call(arguments, 1) : [];
      collectionNames = _.isArray(args[args.length - 1]) ? args.pop() : [];
      dbOptions = args[0] || {};
      url = helper.parseMongoUrl(url);
      connection = new Driver.Connection(url);
      db = connection.db(url.db, dbOptions);
      for (_i = 0, _len = collectionNames.length; _i < _len; _i++) {
        name = collectionNames[_i];
        db[name] = db.collection(name);
      }
      return db;
    }
  });

  require('./connection');

  require('./db');

  require('./collection');

  require('./cursor');

}).call(this);
